core/domain/exp_services_test.py

```diff
@@ -924,24 +1043,66 @@ def setUp(self):
             'param_changes': []
         }
 
-    def test_update_state_name(self):
+    def test_add_state_cmd(self):
+        """ Test adding of states."""
+        exploration = exp_services.get_exploration_by_id(self.EXP_ID)
+
+        self.assertNotIn('new state', exploration.states)
+
+        exp_services.update_exploration(self.owner_id, self.EXP_ID, [{
+            'cmd': exp_domain.CMD_ADD_STATE,
+            'state_name': 'new state',
+        }], 'Add state name')
+
+        exploration = exp_services.get_exploration_by_id(self.EXP_ID)
+        self.assertIn('new state', exploration.states)
+
+    def test_update_state_cmd(self):
```

Comment:
Maybe ``test_rename_state_cmd`` might be more appropriate of a test name; this current name is ambiguous.